core/dbt/parser/schemas.py

```diff
@@ -816,6 +816,24 @@ def parse_patch(
         source_file: SchemaSourceFile = self.yaml.file
         if patch.yaml_key in ['models', 'seeds', 'snapshots']:
             unique_id = self.manifest.ref_lookup.get_unique_id(patch.name, None)
+            if unique_id:
+                resource_type = unique_id.split('.')[0]
+                if (unique_id.split('.')[0] + 's') != patch.yaml_key:
```

Comment:
We need to use [`NodeType.pluralize()`](https://github.com/dbt-labs/dbt/blob/6854e6746406d87e3e800d7e535c85aac5bf6d2e/core/dbt/node_types.py#L52) here, primarily because the plural of `analysis` is `analyses`.

Could we do something like:
```suggestion
                resource_type = NodeType(unique_id.split('.')[0])
                if resource_type.pluralize() != patch.yaml_key:
```

As a guide, [here's](https://github.com/dbt-labs/dbt/blob/745ae3e64ac03fedb19228b31252fea05eb0ed37/core/dbt/contracts/graph/manifest.py#L685-L695) the code in `0.19.latest`... In fact, it looks like we already have an exception for this: [`raise_invalid_patch`](https://github.com/dbt-labs/dbt/blob/6854e6746406d87e3e800d7e535c85aac5bf6d2e/core/dbt/exceptions.py#L927-L942). Perhaps we can turn that into a `warn_or_error`?